src/ui/page/share/dataset/steps/completeTransaction/completeTransaction.tsx

```diff
@@ -0,0 +1,83 @@
+import type { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import * as O from 'fp-ts/Option'
+import { flow, pipe } from 'fp-ts/lib/function'
+import type { FormItem, TextField } from '@/ui/store/slice/shareData/shareData.slice'
+import { Button } from '@/ui/component/button/button'
+import { Icon } from '@/ui/component/icon/icon'
+import { useAppStore } from '@/ui/store'
+import './completeTransaction.scss'
+
+// eslint-disable-next-line max-lines-per-function
+export const CompleteTransaction: FC = () => {
+  const { t, i18n } = useTranslation('share')
+  const locale = i18n.language
+
+  const theme = useAppStore(store => store.theme)
+
+  const { formItems } = useAppStore(state => ({
+    formItems: state.shareData.form
+  }))
+
+  const dataSetTitleField = formItems.find(item => item.type === 'text' && item.title === 'title')
```

Comment:
This should be carried by the slice, through the `query` functions.

In other terms, it's the model responsibility to derive the current state of the applications into a view model.

Reply:
Yes it makes the component cleaner and more focus on rendering.

The field has been retrieved using the `formItemById` query, in [baf88d7](https://github.com/okp4/dataverse-portal/pull/484/commits/baf88d7247a378f41489d3b67955e1ff80ed6257).

Thanks